tests/integration/targets/latest/tasks/plugin_bgp_asn.yml

```diff
@@ -0,0 +1,89 @@
+---
+##
+##
+### PYNAUTOBOT_PLUGIN_BGP_ASN
+##
+##
+- name: "BGP ASN 1: Creation"
+  networktocode.nautobot.plugin:
+    url: "{{ nautobot_url }}"
+    token: "{{ nautobot_token }}"
+    plugin: bgp
+    endpoint: autonomous-systems
+    id:
+      asn: 65001
+    attrs:
+      status: active
+    state: present
+  register: test_one
+
+- name: "BGP ASN 1: ASSERT - Creation"
+  assert:
+    that:
+      - test_one is changed
+      - test_one['diff']['before']['state'] == "absent"
+      - test_one['diff']['after']['state'] == "present"
+      - test_one['autonomous-systems']['asn'] == 65001
+      - test_one['autonomous-systems']['status'] == "active"
+      - test_one['msg'] == "autonomous-systems asn:65001 created"
+
+- name: "BGP ASN 2: Create duplicate"
+  networktocode.nautobot.plugin:
+    url: "{{ nautobot_url }}"
+    token: "{{ nautobot_token }}"
+    plugin: bgp
+    endpoint: autonomous-systems
+    id:
+      asn: 65001
+    attrs:
+      status: active
+    state: present
+  register: test_two
+
+- name: "BGP ASN 2: ASSERT - Create duplicate"
+  assert:
+    that:
+      - test_two is not changed
+      - test_two['autonomous-systems']['asn'] == 65001
+      - test_two['autonomous-systems']['status'] == "active"
+      - test_two['msg'] == "autonomous-systems asn:65001 already exists"
+
+- name: "BGP ASN 3: Update"
+  networktocode.nautobot.plugin:
+    url: "{{ nautobot_url }}"
+    token: "{{ nautobot_token }}"
+    plugin: bgp
+    endpoint: autonomous-systems
+    id:
+      asn: 65001
+    attrs:
+      status: planned
+    state: present
+  register: test_three
+
+- name: "BGP ASN 3: ASSERT - Updated"
+  assert:
+    that:
+      - test_three is changed
+      - test_three['diff']['before']['status'] == "active"
+      - test_three['diff']['after']['status'] == "planned"
+      - test_three['msg'] == "autonomous-systems asn:65001 updated"
+
+- name: "BGP ASN 4: Delete"
```

Comment:
Please add an idempotent delete test to make sure it doesn't state it's deleted again.

Reply:
added